src/State.ts

```diff
@@ -1,22 +1,14 @@
-/* eslint-disable eslint-comments/no-unlimited-disable */
-/* eslint-disable */
-// @ts-nocheck TODO(TS) provide types
-const State = {
+// TODO use State from RNModule
+
+export const State = {
   UNDETERMINED: 0,
   FAILED: 1,
   BEGAN: 2,
   CANCELLED: 3,
   ACTIVE: 4,
   END: 5,
-};
-
-State.print = (state) => {
-  const keys = Object.keys(State);
-  for (let i = 0; i < keys.length; i++) {
-    if (state === State[keys[i]]) {
-      return keys[i];
-    }
-  }
-};
+} as const;
 
-export default State;
+// It is needed for backwards compability, since it can be used as a type and a value
+// eslint-disable-next-line no-redeclare
+export type State = typeof State[keyof typeof State];
```

Comment:
It is taken from `react-native-gesture-handler.d.ts`, so it can be used by users. `typeof State[keyof typeof State];` returns `type State = 0 | 1 | 2 | 3 | 4 | 5`, whereas `typeof State` returns 
```ts
type State = {
    readonly UNDETERMINED: 0;
    readonly FAILED: 1;
    readonly BEGAN: 2;
    readonly CANCELLED: 3;
    readonly ACTIVE: 4;
    readonly END: 5;
}
```

Reply:
I see.
I also tested other variants and they don't work so let's stay with existing types.